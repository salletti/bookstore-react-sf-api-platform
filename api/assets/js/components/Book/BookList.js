import React from "react";
import BookListItem from "./BookListItem";

export default class BookList extends React.Component {
  constructor (props) {
    super(props);
  }

  render () {
    const bookItems = this.props.books.map((book) =>
      <BookListItem
        key={book.id}
        id={book.id}
        title={book.title}
        author={book.author}
        commentsNumber={book.comments.length}
        image={book.image}
      />
    );

    return (
        <div className="col-sm-8 mx-auto">
          <ul className="list-group">
            {bookItems}
          </ul>
        </div>
    );
  }
}
