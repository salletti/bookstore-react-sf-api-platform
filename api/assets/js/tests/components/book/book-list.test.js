import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";

import BookList from "../../../components/Book/BookList";

//mocking component
jest.mock("../../../components/Book/BookListItem", () => {
  return function FakeItem (props) {
    //console.log(props);
    return (
      <li id={props.id} key={props.id}>
        {props.title}
      </li>
    );
  };
});

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const listBook = [
  {
    id: 1,
    title: "title 1",
    author: "author 1",
    comments: ['comment', 'comment'],
    image: '/media_objects/1',
  },
  {
    id: 2,
    title: "title 2",
    author: "author 2",
    comments: ['comment', 'comment'],
    image: '/media_objects/1',
  }
];

it("show should show 2 item", () => {
  act(() => {
    render(
        <BookList books={listBook} />,
        container
    );
  });
  expect(container.querySelector("li").textContent).toBe('title 1');
  expect(container.querySelector("li:nth-child(2").textContent).toBe('title 2');
});

it('item render correctly', () => {
  const tree = renderer
    .create(<BookList books={listBook} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
