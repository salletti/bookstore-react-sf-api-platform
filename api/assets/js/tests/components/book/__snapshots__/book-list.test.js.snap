// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`item render correctly 1`] = `
<div
  className="col-sm-8 mx-auto"
>
  <ul
    className="list-group"
  >
    <li
      id={1}
    >
      title 1
    </li>
    <li
      id={2}
    >
      title 2
    </li>
  </ul>
</div>
`;
